### ISSUE TITLE
Support xml-c14n 1.0 

### ISSUE THREAD

**Issue by andrewstuart** [russellhaering/goxmldsig#4]
So as I've been working on russellhaering/gosaml2#3, I've hit a minor snag with signed AuthN request. It seems that Shibboleth wants signatures that use the [original](https://www.w3.org/TR/xml-c14n) xml-c14n algorithm.

I'd like to try to tackle the issue, but I'm curious as to whether you have any advice/experience on the differences and how I would go about prepping the etree.Document differently. No worries if not, but it might save me some reading of the spec. :-)

**Comment by russellhaering** [comment]
I ran into some issues related to the different canonicalization methods early in the development of this library and mostly just hacked through it and hoped. @phoebesimon improved things some, but I'm not surprised this came up.

The best summary I've seen of the differences between the three methods is here: https://www.w3.org/standards/techs/xmlc14n#w3c_all

We currently claim to implement `xml-c14n11` (which I think we erroneously refer to as exclusive internally). The main differences between `xml-c14n11` and `xml-c14n` seem to relate to attribute inheritance, and at a glance I'm not sure we're actually doing it right for 1.1 so your issue might go a little deeper.

As a side note, I've also been meaning to try and upstream some canonicalization improvements into etree. Most of this probably belongs there.

**Comment by andrewstuart** [comment]
Yeah, it'd definitely be great to upstream it, and thanks for the comparison doc! I eventually ended up reading the spec and playing with the `xmllint` tool to help me grok it better, and canonicalize the xml I used for tests. I have an initial hack step (I called it "prep," which sounds better than "hack" :-P) worked out for xml-exc-c14n, which I've verified at least lets me sign the AuthnRequest and Shib is happy with it.

The two hacks are very similar too, so I'd like to merge them at some point and make it extensible, which would probably be desirable upstream in etree as well.
